### DISCUSSION TITLE
Gfortran and fortls

### DISCUSSION THREAD

**Discussion by davidjayjackson** [fortran-lang/fortls#248] [Q&A]
What is the difference between a server and a compiler?  I have gfortran installed do I still need fortls?

**Comment by gnikit** [comment]
A compiler is responsible for parsing your source code into a binary that your computer can run.

A Language server is a tool meant to provide high level functionality to users while writing code. This functionality was previously reserved only for Integrated Development Environments (IDEs), features like hovering, autocompletions and more, have a look at the fortls' webapge for a demo of the features: https://fortls.fortran-lang.org/features.html
What a language server does it it provides a standardised protocol to communicate this information to multiple code editors like Visual Studio Code, Neovim, Vim, EMACS, Kakoune and so on thus making IDE-like features easily accessible to all developers.

A language server cannot compile or "run" your code, it's just a development tool to make the programmer's life easier.

For more info:
[What is a Compiler](https://www.techtarget.com/whatis/definition/compiler)
[Language Server - Wikipedia](https://en.wikipedia.org/wiki/Language_Server_Protocol)
